olive/systems/docker/docker_system.py

```diff
@@ -171,5 +172,11 @@ def _run_container(self, tempdir, model: OliveModel, metrics: List[Metric], cont
                 )
                 logger.debug("Dev mount cleaned up successfully")
 
+        exit_code = exit_code["StatusCode"]
+        if exit_code != 0:
+            raise docker.errors.ContainerError(
+                container, exit_code, eval_command, self.image, "Docker container evaluation failed"
+            )
+
```

Comment:
better to move this part into the try clause

Reply:
Done.